### PR TITLE
Add list folds to the surface syntax

### PR DESCRIPTION
**Pull request by AltGr**
Temporary syntax:
```
for a initially 0 and then (a + x) for x among lst
```

and in French:
```
pour a initialement 0 puis (a + x) pour x parmi lst
```

to be elaborated with the Syntax Committee of course. Two new keywords: `initially` and `and then` (it could have been the existing `then` keyword for english, relying on the ambiguity of the word, but that's not portable across languages).
Then added to the cheat-sheet and docs.

Note that detuplification takes place, making it possible both to fold multiple lists in parallel, and to use multiple accumulators. See the tests for examples.

### CODE REVIEW

**Comment by AltGr** [comment]
I wonder if non-programmers will be able to make sense of this. Maybe the following conveys the meaning better ?
```
for a initially 0 and then equals (a + x) for x among lst
```

**Comment by AltGr** [comment]
Closes #647

File: tests/array/good/fold.catala_en [hunk 1]
```diff
@@ -1,35 +1,82 @@
...
+declaration insert content list of money
+  depends on element content money,
+             lst content list of money
+  equals
+  let (lower, higher) equals
+    for (lower, higher) initially ([], []) and then
+      (if x <= element then (lower ++ [x], higher) else (lower, higher ++ [x]))
+      for x among lst
+  in
+  lower ++ [element] ++ higher
```
Comment (denismerigoux) [inline tests/array/good/fold.catala_en:33]: HAHAHA that's it we can program insertion sort in Catala, we're doomed !

**Review by denismerigoux — APPROVED** [review]
OK for merging this now, but syntax of fold should be on the agenda of the Catala meeting of Oct 22nd!

File: tests/array/good/fold.catala_en (hunk 1 above)
Comment (AltGr) [reply tests/array/good/fold.catala_en:33]: My thoughts exactly when I was writing it ;)

**Comment by denismerigoux** [comment]
@AltGr did the syntax committee agree with this syntax? Should it be changed?

**Pull request merged**